### PR TITLE
Refactor path resolve

### PR DESCRIPTION
**Pull request by TrySound**

### CODE REVIEW

File: test/import.js [hunk 1]
```diff
@@ -54,12 +55,6 @@ test("should not need `path` option if `source` option has been passed", t => {
   })
 })
 
-test("should be able to consume npm package or local modules", t => {
-  return compareFixtures(t, "modules", {
-    root: ".",
-  })
-})
-
```
Comment (MoOx) [inline test/import.js]: Why did you remove this test(s)?
Looks pretty important to me.

**Comment by TrySound** [comment]
@MoOx What is the case of local npm modules?

**Comment by MoOx** [comment]
web_modules with webpack :)

**Comment by MoOx** [comment]
Note that I think this also cover node_modules.

**Comment by TrySound** [comment]
@MoOx package.json in node_modules or web_modules still works. But it doesn't work in local directories. Also I removed bower_components, caz we still do not support bower.json

**Comment by TrySound** [comment]
@MoOx Okay I saved previous way to resolve paths.

File: index.js [hunk 2]
```diff
@@ -18,7 +18,6 @@ var resolveMedia = require("./lib/resolve-media")
 var moduleDirectories = [
   "web_modules",
   "node_modules",
-  "bower_components",
```
Comment (MoOx) [inline index.js]: This should be added in the CHANGELOG since it's a breaking changes (and we should add how to add bower support like it was).

File: test/fixtures/ignore.expected.css [hunk 3]
```diff
@@ -28,6 +25,10 @@
 
 @import url(//css);
 
+@import "http://css" (min-width: 25em);
+
+@import "http://css-screen" (min-width: 25em) and screen;
+
```
Comment (MoOx) [inline test/fixtures/ignore.expected.css]: Why is the order changed here? This looks like a total weird side effect that will break user expected result...
If the ignored file is being imported first, imported content should be imported at this place, not later.
Looks like we are introducing a bug :)

File: test/import.js [hunk 4]
```diff
@@ -155,14 +154,16 @@ test("should work with no styles without throwing an error", t => {
...
   const resolve = require("resolve")
-  const sassResolve = (file, opts) => {
-    opts = opts || {}
-    opts.extensions = [ ".scss", ".css" ]
-    opts.packageFilter = pkg => {
-      pkg.main = pkg.sass || pkg.style || "index"
-      return pkg
-    }
-    return resolve.sync(file, opts)
+  const sassResolve = (id, base, opts) => {
```
Comment (MoOx) [inline test/import.js]: if the signature of the callback to provide custom resolve, we should also document that and specify this in the CHANGELOG.

**Comment by TrySound** [comment]
@MoOx Done!

**Comment by TrySound** [comment]
Is PR too big or you can accept it?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,9 @@
 - Removed: async mode/option
 ([#107](https://github.com/postcss/postcss-import/pull/107))
+- Removed: bower is not currently supported, use custom resolver
+([#116](https://github.com/postcss/postcss-import/pull/116))
```
Comment (MoOx) [inline CHANGELOG.md]: fyi, you can just add bower support by it to the `path` option.
So I would use ``"bower_components" not supported by default anymore, use "path" option to add it back``.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,9 @@
...
+- Changed: custom resolve has more responsibility for paths resolving
```
Comment (MoOx) [inline CHANGELOG.md]: This message is not clear. We should explicitly tell what has changed in the signature of the function (or send the user to the right place in the doc).

File: CHANGELOG.md (hunk 5 above)
Comment (TrySound) [inline CHANGELOG.md]: bower support is also bower.json support, which is not simple with main field as array.

**Comment by MoOx** [comment]
@TrySound Except my last comments about the CHANGELOG, this PR looks good. Great job. I just want to be sure that the developer experience for upgrading is easy :D

File: CHANGELOG.md (hunk 6 above)
Comment (TrySound) [inline CHANGELOG.md]: Not it just gets `uri` and `baseDir` and should return resolved path.

File: CHANGELOG.md (hunk 5 above)
Comment (MoOx) [inline CHANGELOG.md]: I know that, but bower is more a downloader, so I guess lots of people just point to the right css file when they import it (eg: `@import "jquery-slider-fx-2016/slider.css"`) so we don't really care about it. The idea is just to inform people that are currently using the feature we are breaking how to upgrade easily.

File: CHANGELOG.md (hunk 6 above)
Comment (MoOx) [inline CHANGELOG.md]: So you can add at the end of the messsage: "See XXX for more information about this change" to point to the right place in the README.

**Comment by TrySound** [comment]
@MoOx Done

**Comment by MoOx** [comment]
:+1:
